packages/common/src/store/ui/stripe-modal/types.ts

```diff
@@ -16,3 +16,21 @@ export type StripeModalState = {
   stripeSessionStatus?: StripeSessionStatus
   stripeClientSecret?: string
 }
+
+export type StripeSessionCreationErrorResponseData = {
+  error: string
+  code: string
+  message: string
+  type: string
+}
+
+export class StripeSessionCreationError extends Error {
+  constructor(
+    public code: string,
+    // Avoiding `message` to not shadow the `message` property on Error
```

Comment:
nice, good catch. maybe should be stripeErrorMessage?

Reply:
Yeah that works too. I think I was going for a "generic" pattern we could use in the future. But that's not really necessary and your suggestion makes it clear where the error comes from.